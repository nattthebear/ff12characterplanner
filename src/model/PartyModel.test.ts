import { describe, snapshot } from "../../test/snapshots";
import { Boards } from "../data/Boards";
import { Licenses } from "../data/Licenses";
import PartyModel from "./PartyModel";

// A simple deterministic random number generator to provide for snapshot tests
class XorShift32 {
	private state: number;

	constructor(seed: number) {
		seed = seed | 0;
		if (!seed) {
			throw new Error("Seed must be non-zero");
		}
		this.state = seed;
	}

	next() {
		let x = this.state;
		x ^= x << 13;
		x ^= x >>> 17;
		x ^= x << 5;
		return (this.state = x) >>> 0;
	}
}

function doTest(r: XorShift32, count: number) {
	let p = new PartyModel();
	const c = r.next() % 6;
	p = p.addJob(c, Boards[r.next() % 12]);
	p = p.addJob(c, Boards[r.next() % 12]);
	const ll = Licenses.length;
	for (let i = 0; i < count; i++) {
		if (r.next() % 2) {
			p = p.delete(c, Licenses[r.next() % ll]);
		} else {
			p = p.add(c, Licenses[r.next() % ll]);
		}
	}
	return p.encode().split(".")[c];
}

describe("PartyModel", import.meta.url, it => {
	it("basic stress tests", () => {
		const r = new XorShift32(8675309);
		for (let i = 0; i < 200; i++) {
			snapshot(doTest(r, 2000));
		}
	});

	it("timed tests", () => {
		const r = new XorShift32(69420);
		const start = performance.now();
		for (let i = 0; i < 350; i++) {
			doTest(r, 1600);
		}
		snapshot(performance.now() - start);
	});
});
