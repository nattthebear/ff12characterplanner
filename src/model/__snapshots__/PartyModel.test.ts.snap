{
	"PartyModel": {
		"basic stress tests": [
			"T4b4g4sDAAaAAAAAAAAAAAAAfyBCiACA",
			"JgIA1vx1Qoh3AJgjQPWDiQgjGAIgmCKA",
			"D5AF8nDsCzGuHJ0QTs_obeND8kugMxE",
			"IdmEOObjyfSQ5j3251d-hm4wjmUUcg",
			"ZILXFp305tjYI5SHrOfPZeiQ2d0pj7h_",
			"OwCA-AcDADIw_COWxxAIGh04BABgAQE",
			"IpjMoGIdQ5X-kfjDIAC3Qo-LCETefJblGg",
			"f8nQ3Kuv7NvyC58DYBM4gJFl2LMkSrgpWg",
			"lgAAAAAA5IEHAAAAAAAAAAAAAAZA",
			"UWAtuDcBCGLdUACwCN8HAF0HESCQ",
			"J0a3Lt7EF2L42MA60ANhIMgKE4IAmwM",
			"GIMEgAQshw0xnM9BYHchQMDBQID8AAAgyQ",
			"jgAECNjJdH794euAIQjwDgSnKSHiEA",
			"gKrB_e-1ipLgSRAQANhnYqADAgwAAcED",
			"gNThrnVzN1IKFPCawV-gtvnFc4g8QP8B",
			"iwABowkBYCEofF8TFAAAADjQ566aSA0",
			"MMrzMPPmYmZneEigQKABACMCigAE",
			"nGgBA2AY_sUJQAAf0IQW-0cTACQA-SQc",
			"JxQYsY3cSBfnMMIFCKzbUEoAgIOzJhY",
			"ZMr-MxxADhoLAzCiRW5IAAAAcAaQEAAE",
			"R8oVvh55E16uwE0O7cjDYQ9MeC48oHGA",
			"UuXwAAMgsAgPcNbAABFABTJGAJCMZDhQBIIh",
			"aJDXgf17FRcUOMShglZ8wBVMwAgmQCA",
			"l2BPwQcghqtx_jmhJQggXSMRfGQr",
			"aAAI0BggCAjAEAQABAbAAAAAAAAg",
			"MGJxwAficBMiyQP2nCDjzAvgCABgAQ",
			"Ls33UwbWkowk2BQPfBj3MehJB4hwUIgz",
			"nAAwAEQc-uUxE2IgC94WHuAJABiC-Ag",
			"QIEAfgAYHN7wX7CBYAe7hD0QAIGoz1HgBg",
			"f2lA3aVpb9_nCxsEjB_YQzQnMNKEPZChIAQD",
			"G8gMBCAQ7ByEC6TzBTRABB_gc2AJJAE",
			"aaA-M4hgDg8BLDADA1sEYxAEsopwjSE",
			"ZgAAgAkiPDQQAAhAIAAAAOAAACI",
			"iwC2XiwtY3_nQMQJLDTYLANYoBXqHAMB",
			"DQD4-IQ9A3egABwYZHMEhvyE",
			"eQAgIp-389viBQoCgAM",
			"ckCOAXA0gWUT89kXBUo-gWAEBJAIvAAB",
			"mEjDh7EU_tU_9PBHEESjwzkMEMOCHSAkQA",
			"GgBG_AFwuqYwBOjnD2rmrHDKgoCJLeB6Iw",
			"IoOBoAcAc_i2w1njED92TgEIBmDDfEERBQ",
			"cWIAAEAgBmoAwAcsNAAAAAAgAChOQAAAAQ",
			"kAAAgACFMyILUOUpb7jgEBQAAIAfICg",
			"m8AYx0MABuQAAeQTVhphM-MgAICUJAAgDIQ",
			"FeQDrDn0AVAA8z04xlb4O6-Bgh0e_P9n",
			"EwAQAAYADhCDikl0bOYUFAQEEAAAEAID",
			"Fvrt4Is_j4bcJIhx4tLD4f85MARAsAE",
			"PQAA-f9iPl78mnzcEZAARAAC",
			"RA4AAiB5O03wRSCpKOLj5aAQEBHwAg",
			"VwjAaxziEDmIMbs1ZEAJeYsZFVHYzwI",
			"GoBBoEdouAEMAOyGfCamIoDYwmd8CAQK",
			"XMBC-oYHgMCCNIEEAAAIAA4QtGxB5Bk",
			"KwE47z3CllPcPwgbEHYhAkO5jJQCKyA",
			"Z6WMCw5uPhYGGMAAdMCABgYQeAIBAAAE",
			"PQAA-IciElJw4sQEEYAD",
			"bAqDwQxQGjZNdvBNFwtciAIAAEAYgAo",
			"WtALzIpwQMZPMtKPDgAA_AQmTAMZRAQD",
			"dByMAAAgBGIDAAHryED8IwAATGJCBAE",
			"RggBCIAgEBhitAgQQIBDgLAAAABA6OODAsAB",
			"NgEwEAJ-ADVHlG38YwACQAIhEHsVPQMAAEAgAg",
			"PUYC_uMhNT3wsUICDVAABng",
			"EwEKAA-sBibCIsgp3sLxf5gBhzAtEJIB",
			"QtgdDsTgW3RpdwQ2WwETwAVY8GwEEIiA",
			"mBAJhMA8_MPwQwSHgUmLi3MAAONJADiAQQ",
			"IwEAAAAAWAAAAAyGBMjBAYaggNc_VQI",
			"cXIghssOJuIB_ASusoXOvwAAACBOkEKm5w",
			"LwB4hSNx8gOC48H7kB_AAcMkLYvz",
			"EAgBGyNnCmPyhqYUzLJMuZObC_4JEww",
			"cQgYd_o4BPaDnwmGL4BHOg8AAAy2DwTgIgk",
			"VhLd0JYmrMaFMvSAhWsWX0EAAjoDq0cQ",
			"c92ddD-my-wbPhlviFAkCASYsFM",
			"RSxcnOfSBifo6XDAATjjmRB0EKZ3g6GfCA",
			"Uvy9g4dBwnKBBIAAAAwMhNwBAG3OdTxDPAgS",
			"PzB8vrIrFWx4-IQAHAADAeBgABCAAgE",
			"JnJABHTBe4IwjAAfsB7jCEBkGIABTA",
			"NaVH4A79gzGYC_ICXC0GAAQGAMQZRAAAhAg",
			"YxG3Hp9sOgJRQmQuhKJHBKwYABDlBg",
			"kQABBGAMfKgAfDABKIBw",
			"NgV0CAANQDQQPgAQACAQAAMgPoAEAACBoAUg",
			"eSLAPjcn48_nBQgOsAP8MA",
			"ZQDAYAwlHwxepLwBRGJOUKIRAAAQPkOAAw",
			"bQCAcSEwBn8MlB8GBcgwEA",
			"a5B-bJ4FHmdnBnwgghJAAG4UxUzvZ9E",
			"cj3N6gSeEUAehl3CApCAh5LeiPBKPQ6T",
			"EeIAtOAO_KMSB0E4enYNzPghYTNAVt4ABw",
			"DkADPAAEAAAIwkVYxEUpLGBAAAASADQG",
			"HhAIiMIhOKAAVvjGIAfswGDgMQIAAg",
			"KcDAPXztx4DZ_EoC9GuCRQEIjlxAEAg-hA",
			"lwIEYNExf7cA8S7HOUiRXfAhZCRHBA",
			"ikAlCIzOPFAIBwsA8eODACgQwMciiCDU",
			"bgAAAQDAx3sQwEEMCIB9NALAgz8IAAAAEA",
			"O1iO5D5ANDc3fqH9YAgoeFjFBgFDNvEC",
			"by2M4GMgGmg8sHSQcSIIBMKAjAJEZQ",
			"FQEweIENCAGBhLMpbqRpDAABwjeQO3hGBA",
			"PlpPCjxqEhq8ZHnYWbbBHlAAIARgHgRYiwQ",
			"KcCzsSxs4Oy72NgvUSneeADgCm1nAwYG2AE",
			"FFftc0OoUgBMw8Fw3gKNCdwBgBJstaZABgY",
			"M4W3C3g-AjW3k-seZCy4NzhPD7FA-EImwCA",
			"M4W-BwgJ8RqQmVVIZggywEB_l0MECMQcv0c",
			"cABznO0wxuEa7H9sDdBm0pqc7gg",
			"VyIR3wAEmG1XIeY9gQAC9iwAADCKWQw",
			"PQAAAIAgEBwwwMAAAzAAD3w",
			"eIx7KnnatnU2uj1_cNDwKwRWpD4",
			"b3BAcWWo8MccMXas9Wg-HCdACtBIJA",
			"eYCBmAUFBgYGCPgmABt_CgY",
			"JkIAVL7Ha2KOj8UT8PNfLoho8QER0AFg",
			"cUDIAQ4ExO8QwWOOcNx-LQAAAACoBxgBhAY",
			"TFzF87HBWxDaZIDBFCKBYICACQIQ",
			"i8gnqx7P5kzh4eRGeOvUO0TMllAxOVK0",
			"Nhgc-D8aEjg734-cIwQHfGMkkLEKUowQ",
			"fsGAGYvn69XgARgGABgwYcACYOCBI4SQcBA",
			"F8acEOoHqhyOD-Z44vc8L2wMNWZBgA",
			"kgAAxnAEBoC746LqDtP4I9ZIAu1zgkMp8w8",
			"awIDBB4IgsoCBCIEBeJAAwQLAESOhw",
			"bij3M1eDDso8tE08vDu-CQz9a4glLz4AEA",
			"ZkAhIcxu5pQHHuAUD1AjF-EQJUqH",
			"lBIAAEAMDuBzYCARCACBoQPOAAIQ",
			"L0A2GBrCUGLAIAggOAA5AuIAerWPBg",
			"DQABcIAbAOagA3Q45IIYHPgP",
			"Yi-tG38cBroYUzxnASAADhwIAySGDEasBAM",
			"hEEzmI1FAwwECBAuQDgGBAEmgDlFPweA5IE",
			"bgAAQEVygv8UwH8MQJh-KBMJgVxJKvgKGA",
			"hQBAyA-D8AcVfgkIJThYDygc",
			"cQACRow1B1QR6F2sMwS_PQKg8Tg4BQAAwRA",
			"l0KJg2FtCrngrJEE1OAQw2EoNIq7",
			"DRV7fIH4AAIAAAAAYOEIH3REAg",
			"bSB-cOOzgv88Ij8cDEggEA",
			"QwAgAAEAABAgODAAAAAAAAAAMAAQ",
			"aIBAR59HPwwEaBkQAAABAt4qwNksiQI",
			"lYCBiIAd_MM6_RFAAEiZO4cAhAIAxj0pAEQD",
			"dmA4-0XCzHEzwHvuUz4APNgwtToDAEwiAw",
			"acLtq8wjaGg4UOEAh5mFNeAEwcNwjy0M",
			"XeQCaojAYGvs8FCCARAB8GAAAISc9o84",
			"HgQOiBTPQAM-2BwnEyIBCHAT-ooAFA",
			"GQAAgE2uCdE_jtwGAgIQ",
			"mUCBDUAHDKIH4OAPAICIMBguZM8gAAg",
			"icqIgwNDNy_IB4wEbcbd7mxHGjiO2dcO",
			"IjiIrEYJA2aYLwwGGP86XmhApHLL-DBM",
			"EQBC4IAPDgLLgDU8bGs7gMABQApKBhoVgQ",
			"EABiOAbcAWmyI_wYQuP7IZgBCYAR",
			"bUDgMO00B_yBPwXODc9-CA8",
			"eIQHADEIguPjQvfn7ozgAwEAFCM",
			"Y4BEUBhACgoCBAAAAABAEIAAQJDO",
			"OEUeWAB4ABEAoYIQCTECQ8BHAIBxAAAQBA",
			"hgAc8KAFdWIAAAAOROAEEChQTo3bLw",
			"LUDDVrDuApiTHObf5CvADwAUFgE",
			"jIAUhQGisnVdoe8DABysDR4ehCIiqEIF",
			"dkTa9n-A5vIbRzQMNiYBIAAAAHMGAjWG",
			"a4hGer5ABk58CmYhjjMsA3yKzMHAe_kH",
			"WwB4AWx0gELgoKK6D15ZSwYBexIhEkAAsA",
			"bQBAMCsUB3wAED8PBUswDgE",
			"TTQ8gHfDBzqAi8CTQMoDYogJABwAjhvCDagf",
			"XYBBCAABbmgOsDiiIhEBeAABAAYAD8QR",
			"goAAmIDNgAAEEJBxABlAAIyKEYyhCf2qAg",
			"XP243AqFVNXPqjcQAQAAAAAAIgOQSaA",
			"RM7jhAfA96G168jM6Gmf-jsA8MkXCg",
			"UmTxC8kAOjqAnoaQQcMBAAwARBKEICAQIQgI",
			"JqBfrM7JcrZU352DgCHCGQAFCPeZygPI",
			"J4xqMbbmGHDwGITq7O7hCwcAFJEjQyA",
			"ggADgQGDw57gCwA8cAwHQAIEhIgAOEAwBg",
			"P0MH4oAweIS-VLCBzTfQCAgAACSQAg",
			"lM69H_XE_vAhzFXbyQUZf3F58ZFk",
			"PSDA-IcpF144_lQEGVgAPw",
			"d61AAlxGTmozz0eKap3CIwgAgnAwAQw",
			"Dy2AIBBnCJ3yAhmhwdAYjsgB95d1MfEB",
			"joR52hOI8MFiAceDowGAAAaMxbuHwCM",
			"fgCA-COj6vDwhRgEIAcMAYKTAAAAYgfIGQI",
			"lmAAAxAIDsTjgABEwJsDAAAACALAAMIB",
			"e-B5gIMPC4EBAXAAoBs8gY3qPACwrwEKDA",
			"DwsQEYa7ioDTgwHkTewNfBwQwB9KcxND",
			"XEob_TIFHFrw7h0AAhID848Y9B_kpQk",
			"cyCEYOIQBvgBAGEEBkgwCAcW",
			"UKBwgZ9DAwARUJAwUGAAqAMAQkDi",
			"MH_nPl3hEONALFpOwkzPoO9geDcUAQ",
			"Lkg_vABplh8vCAYIAAAACgKS2s08AAAQ",
			"N5Fc-iCcvzjfycNAUQ4Hv-ZkcEniqoJA",
			"KeBQR_zh86sBCIQPBCjAAgAAAExEEjAEAQ",
			"kwYBAIBh9Pg3JIANeIiRuAAEDIBDJw",
			"lQAACnIFKu8_QAADAEAIGEAAjH4cxBAB",
			"g8mjUQPJZf1BAtB86RDAwQwRRKAY0AE",
			"iGAAgEMEs91_92oUAAAAAAAAAAIACQUC",
			"TYTN8c8GBwiAeo44yHyEi6gO7I9ALAdSBSSYAg",
			"U4DR0csKhAkAHpTJAgOJEYAAAIHIfxE",
			"SuQ9gdNQj_qFxcAw-HQsDl0Mg9i8gggQ",
			"YIAwCg5miDgOcOru0AEwLqkAEJ77KUGZwg",
			"NglwQAB4ADMzvCHoYwwGcMBIFFk2MQ",
			"WID83J4nzCLmcfKnEdpaX_UhBBudTzM",
			"HcBGEAAwbjPgyyx_AUJ0fkFA0OAN",
			"hswv2SKH8hfkY6CAAOSnOHtYAWddAgE",
			"bIfVmY9bDjoTPlBBjk_FS9MGAUAYIJAc",
			"UaYuzP1tPw2fM54AEjgdIAmAuCUYAg",
			"IEAMgEAgOAAMABAGAoQjjwEUAAJBAg",
			"YyAIkRlwDhQcAAAAAAAAAABEoIEO",
			"J5VTmXngoYy6CIACQKvPiAAIgIelCxc",
			"GgFIzygxOJ4x1sWHMd7zfj-pDEZMcrAYEA",
			"a4ELHg9_ByYQA5CBBMJEAQICACAO5RM",
			"ZwEIHAwAHgwGEAAMAKBAABBAfCc",
			"c393uD5gx-vPiRQeCJAFEwsYA0AC",
			"Sr6cgMMz7aADOgICIwychEEBYgSCQkCFCA",
			"RsADBgDgNBC2pDBAQChig3EBAADkpOEZTNAD",
			"OxBw3J9NpjAziQHcZRwCQwA9MgLInAAa"
		],
		"timed tests": [
			"T4b4g4sDAAaAAAAAAAAAAAAAfyBCiACA",
			"JgIA1vx1Qoh3AJgjQPWDiQgjGAIgmCKA",
			"D5AF8nDsCzGuHJ0QTs_obeND8kugMxE",
			"IdmEOObjyfSQ5j3251d-hm4wjmUUcg",
			"ZILXFp305tjYI5SHrOfPZeiQ2d0pj7h_",
			"OwCA-AcDADIw_COWxxAIGh04BABgAQE",
			"IpjMoGIdQ5X-kfjDIAC3Qo-LCETefJblGg",
			"f8nQ3Kuv7NvyC58DYBM4gJFl2LMkSrgpWg",
			"lgAAAAAA5IEHAAAAAAAAAAAAAAZA",
			"UWAtuDcBCGLdUACwCN8HAF0HESCQ",
			"J0a3Lt7EF2L42MA60ANhIMgKE4IAmwM",
			"GIMEgAQshw0xnM9BYHchQMDBQID8AAAgyQ",
			"jgAECNjJdH794euAIQjwDgSnKSHiEA",
			"gKrB_e-1ipLgSRAQANhnYqADAgwAAcED",
			"gNThrnVzN1IKFPCawV-gtvnFc4g8QP8B",
			"iwABowkBYCEofF8TFAAAADjQ566aSA0",
			"MMrzMPPmYmZneEigQKABACMCigAE",
			"nGgBA2AY_sUJQAAf0IQW-0cTACQA-SQc",
			"JxQYsY3cSBfnMMIFCKzbUEoAgIOzJhY",
			"ZMr-MxxADhoLAzCiRW5IAAAAcAaQEAAE",
			"R8oVvh55E16uwE0O7cjDYQ9MeC48oHGA",
			"UuXwAAMgsAgPcNbAABFABTJGAJCMZDhQBIIh",
			"aJDXgf17FRcUOMShglZ8wBVMwAgmQCA",
			"l2BPwQcghqtx_jmhJQggXSMRfGQr",
			"aAAI0BggCAjAEAQABAbAAAAAAAAg",
			"MGJxwAficBMiyQP2nCDjzAvgCABgAQ",
			"Ls33UwbWkowk2BQPfBj3MehJB4hwUIgz",
			"nAAwAEQc-uUxE2IgC94WHuAJABiC-Ag",
			"QIEAfgAYHN7wX7CBYAe7hD0QAIGoz1HgBg",
			"f2lA3aVpb9_nCxsEjB_YQzQnMNKEPZChIAQD",
			"G8gMBCAQ7ByEC6TzBTRABB_gc2AJJAE",
			"aaA-M4hgDg8BLDADA1sEYxAEsopwjSE",
			"ZgAAgAkiPDQQAAhAIAAAAOAAACI",
			"iwC2XiwtY3_nQMQJLDTYLANYoBXqHAMB",
			"DQD4-IQ9A3egABwYZHMEhvyE",
			"eQAgIp-389viBQoCgAM",
			"ckCOAXA0gWUT89kXBUo-gWAEBJAIvAAB",
			"mEjDh7EU_tU_9PBHEESjwzkMEMOCHSAkQA",
			"GgBG_AFwuqYwBOjnD2rmrHDKgoCJLeB6Iw",
			"IoOBoAcAc_i2w1njED92TgEIBmDDfEERBQ",
			"cWIAAEAgBmoAwAcsNAAAAAAgAChOQAAAAQ",
			"kAAAgACFMyILUOUpb7jgEBQAAIAfICg",
			"m8AYx0MABuQAAeQTVhphM-MgAICUJAAgDIQ",
			"FeQDrDn0AVAA8z04xlb4O6-Bgh0e_P9n",
			"EwAQAAYADhCDikl0bOYUFAQEEAAAEAID",
			"Fvrt4Is_j4bcJIhx4tLD4f85MARAsAE",
			"PQAA-f9iPl78mnzcEZAARAAC",
			"RA4AAiB5O03wRSCpKOLj5aAQEBHwAg",
			"VwjAaxziEDmIMbs1ZEAJeYsZFVHYzwI",
			"GoBBoEdouAEMAOyGfCamIoDYwmd8CAQK",
			"XMBC-oYHgMCCNIEEAAAIAA4QtGxB5Bk",
			"KwE47z3CllPcPwgbEHYhAkO5jJQCKyA",
			"Z6WMCw5uPhYGGMAAdMCABgYQeAIBAAAE",
			"PQAA-IciElJw4sQEEYAD",
			"bAqDwQxQGjZNdvBNFwtciAIAAEAYgAo",
			"WtALzIpwQMZPMtKPDgAA_AQmTAMZRAQD",
			"dByMAAAgBGIDAAHryED8IwAATGJCBAE",
			"RggBCIAgEBhitAgQQIBDgLAAAABA6OODAsAB",
			"NgEwEAJ-ADVHlG38YwACQAIhEHsVPQMAAEAgAg",
			"PUYC_uMhNT3wsUICDVAABng",
			"EwEKAA-sBibCIsgp3sLxf5gBhzAtEJIB",
			"QtgdDsTgW3RpdwQ2WwETwAVY8GwEEIiA",
			"mBAJhMA8_MPwQwSHgUmLi3MAAONJADiAQQ",
			"IwEAAAAAWAAAAAyGBMjBAYaggNc_VQI",
			"cXIghssOJuIB_ASusoXOvwAAACBOkEKm5w",
			"LwB4hSNx8gOC48H7kB_AAcMkLYvz",
			"EAgBGyNnCmPyhqYUzLJMuZObC_4JEww",
			"cQgYd_o4BPaDnwmGL4BHOg8AAAy2DwTgIgk",
			"VhLd0JYmrMaFMvSAhWsWX0EAAjoDq0cQ",
			"c92ddD-my-wbPhlviFAkCASYsFM",
			"RSxcnOfSBifo6XDAATjjmRB0EKZ3g6GfCA",
			"Uvy9g4dBwnKBBIAAAAwMhNwBAG3OdTxDPAgS",
			"PzB8vrIrFWx4-IQAHAADAeBgABCAAgE",
			"JnJABHTBe4IwjAAfsB7jCEBkGIABTA",
			"NaVH4A79gzGYC_ICXC0GAAQGAMQZRAAAhAg",
			"YxG3Hp9sOgJRQmQuhKJHBKwYABDlBg",
			"kQABBGAMfKgAfDABKIBw",
			"NgV0CAANQDQQPgAQACAQAAMgPoAEAACBoAUg",
			"eSLAPjcn48_nBQgOsAP8MA",
			"ZQDAYAwlHwxepLwBRGJOUKIRAAAQPkOAAw",
			"bQCAcSEwBn8MlB8GBcgwEA",
			"a5B-bJ4FHmdnBnwgghJAAG4UxUzvZ9E",
			"cj3N6gSeEUAehl3CApCAh5LeiPBKPQ6T",
			"EeIAtOAO_KMSB0E4enYNzPghYTNAVt4ABw",
			"DkADPAAEAAAIwkVYxEUpLGBAAAASADQG",
			"HhAIiMIhOKAAVvjGIAfswGDgMQIAAg",
			"KcDAPXztx4DZ_EoC9GuCRQEIjlxAEAg-hA",
			"lwIEYNExf7cA8S7HOUiRXfAhZCRHBA",
			"ikAlCIzOPFAIBwsA8eODACgQwMciiCDU",
			"bgAAAQDAx3sQwEEMCIB9NALAgz8IAAAAEA",
			"O1iO5D5ANDc3fqH9YAgoeFjFBgFDNvEC",
			"by2M4GMgGmg8sHSQcSIIBMKAjAJEZQ",
			"FQEweIENCAGBhLMpbqRpDAABwjeQO3hGBA",
			"PlpPCjxqEhq8ZHnYWbbBHlAAIARgHgRYiwQ",
			"KcCzsSxs4Oy72NgvUSneeADgCm1nAwYG2AE",
			"FFftc0OoUgBMw8Fw3gKNCdwBgBJstaZABgY",
			"M4W3C3g-AjW3k-seZCy4NzhPD7FA-EImwCA",
			"M4W-BwgJ8RqQmVVIZggywEB_l0MECMQcv0c",
			"cABznO0wxuEa7H9sDdBm0pqc7gg",
			"VyIR3wAEmG1XIeY9gQAC9iwAADCKWQw",
			"PQAAAIAgEBwwwMAAAzAAD3w",
			"eIx7KnnatnU2uj1_cNDwKwRWpD4",
			"b3BAcWWo8MccMXas9Wg-HCdACtBIJA",
			"eYCBmAUFBgYGCPgmABt_CgY",
			"JkIAVL7Ha2KOj8UT8PNfLoho8QER0AFg",
			"cUDIAQ4ExO8QwWOOcNx-LQAAAACoBxgBhAY",
			"TFzF87HBWxDaZIDBFCKBYICACQIQ",
			"i8gnqx7P5kzh4eRGeOvUO0TMllAxOVK0",
			"Nhgc-D8aEjg734-cIwQHfGMkkLEKUowQ",
			"fsGAGYvn69XgARgGABgwYcACYOCBI4SQcBA",
			"F8acEOoHqhyOD-Z44vc8L2wMNWZBgA",
			"kgAAxnAEBoC746LqDtP4I9ZIAu1zgkMp8w8",
			"awIDBB4IgsoCBCIEBeJAAwQLAESOhw",
			"bij3M1eDDso8tE08vDu-CQz9a4glLz4AEA",
			"ZkAhIcxu5pQHHuAUD1AjF-EQJUqH",
			"lBIAAEAMDuBzYCARCACBoQPOAAIQ",
			"L0A2GBrCUGLAIAggOAA5AuIAerWPBg",
			"DQABcIAbAOagA3Q45IIYHPgP",
			"Yi-tG38cBroYUzxnASAADhwIAySGDEasBAM",
			"hEEzmI1FAwwECBAuQDgGBAEmgDlFPweA5IE",
			"bgAAQEVygv8UwH8MQJh-KBMJgVxJKvgKGA",
			"hQBAyA-D8AcVfgkIJThYDygc",
			"cQACRow1B1QR6F2sMwS_PQKg8Tg4BQAAwRA",
			"l0KJg2FtCrngrJEE1OAQw2EoNIq7",
			"DRV7fIH4AAIAAAAAYOEIH3REAg",
			"bSB-cOOzgv88Ij8cDEggEA",
			"QwAgAAEAABAgODAAAAAAAAAAMAAQ",
			"aIBAR59HPwwEaBkQAAABAt4qwNksiQI",
			"lYCBiIAd_MM6_RFAAEiZO4cAhAIAxj0pAEQD",
			"dmA4-0XCzHEzwHvuUz4APNgwtToDAEwiAw",
			"acLtq8wjaGg4UOEAh5mFNeAEwcNwjy0M",
			"XeQCaojAYGvs8FCCARAB8GAAAISc9o84",
			"HgQOiBTPQAM-2BwnEyIBCHAT-ooAFA",
			"GQAAgE2uCdE_jtwGAgIQ",
			"mUCBDUAHDKIH4OAPAICIMBguZM8gAAg",
			"icqIgwNDNy_IB4wEbcbd7mxHGjiO2dcO",
			"IjiIrEYJA2aYLwwGGP86XmhApHLL-DBM",
			"EQBC4IAPDgLLgDU8bGs7gMABQApKBhoVgQ",
			"EABiOAbcAWmyI_wYQuP7IZgBCYAR",
			"bUDgMO00B_yBPwXODc9-CA8",
			"eIQHADEIguPjQvfn7ozgAwEAFCM",
			"Y4BEUBhACgoCBAAAAABAEIAAQJDO",
			"OEUeWAB4ABEAoYIQCTECQ8BHAIBxAAAQBA",
			"hgAc8KAFdWIAAAAOROAEEChQTo3bLw",
			"LUDDVrDuApiTHObf5CvADwAUFgE",
			"jIAUhQGisnVdoe8DABysDR4ehCIiqEIF",
			"dkTa9n-A5vIbRzQMNiYBIAAAAHMGAjWG",
			"a4hGer5ABk58CmYhjjMsA3yKzMHAe_kH",
			"WwB4AWx0gELgoKK6D15ZSwYBexIhEkAAsA",
			"bQBAMCsUB3wAED8PBUswDgE",
			"TTQ8gHfDBzqAi8CTQMoDYogJABwAjhvCDagf",
			"XYBBCAABbmgOsDiiIhEBeAABAAYAD8QR",
			"goAAmIDNgAAEEJBxABlAAIyKEYyhCf2qAg",
			"XP243AqFVNXPqjcQAQAAAAAAIgOQSaA",
			"RM7jhAfA96G168jM6Gmf-jsA8MkXCg",
			"UmTxC8kAOjqAnoaQQcMBAAwARBKEICAQIQgI",
			"JqBfrM7JcrZU352DgCHCGQAFCPeZygPI",
			"J4xqMbbmGHDwGITq7O7hCwcAFJEjQyA",
			"ggADgQGDw57gCwA8cAwHQAIEhIgAOEAwBg",
			"P0MH4oAweIS-VLCBzTfQCAgAACSQAg",
			"lM69H_XE_vAhzFXbyQUZf3F58ZFk",
			"PSDA-IcpF144_lQEGVgAPw",
			"d61AAlxGTmozz0eKap3CIwgAgnAwAQw",
			"Dy2AIBBnCJ3yAhmhwdAYjsgB95d1MfEB",
			"joR52hOI8MFiAceDowGAAAaMxbuHwCM",
			"fgCA-COj6vDwhRgEIAcMAYKTAAAAYgfIGQI",
			"lmAAAxAIDsTjgABEwJsDAAAACALAAMIB",
			"e-B5gIMPC4EBAXAAoBs8gY3qPACwrwEKDA",
			"DwsQEYa7ioDTgwHkTewNfBwQwB9KcxND",
			"XEob_TIFHFrw7h0AAhID848Y9B_kpQk",
			"cyCEYOIQBvgBAGEEBkgwCAcW",
			"UKBwgZ9DAwARUJAwUGAAqAMAQkDi",
			"MH_nPl3hEONALFpOwkzPoO9geDcUAQ",
			"Lkg_vABplh8vCAYIAAAACgKS2s08AAAQ",
			"N5Fc-iCcvzjfycNAUQ4Hv-ZkcEniqoJA",
			"KeBQR_zh86sBCIQPBCjAAgAAAExEEjAEAQ",
			"kwYBAIBh9Pg3JIANeIiRuAAEDIBDJw",
			"lQAACnIFKu8_QAADAEAIGEAAjH4cxBAB",
			"g8mjUQPJZf1BAtB86RDAwQwRRKAY0AE",
			"iGAAgEMEs91_92oUAAAAAAAAAAIACQUC",
			"TYTN8c8GBwiAeo44yHyEi6gO7I9ALAdSBSSYAg",
			"U4DR0csKhAkAHpTJAgOJEYAAAIHIfxE",
			"SuQ9gdNQj_qFxcAw-HQsDl0Mg9i8gggQ",
			"YIAwCg5miDgOcOru0AEwLqkAEJ77KUGZwg",
			"NglwQAB4ADMzvCHoYwwGcMBIFFk2MQ",
			"WID83J4nzCLmcfKnEdpaX_UhBBudTzM",
			"HcBGEAAwbjPgyyx_AUJ0fkFA0OAN",
			"hswv2SKH8hfkY6CAAOSnOHtYAWddAgE",
			"bIfVmY9bDjoTPlBBjk_FS9MGAUAYIJAc",
			"UaYuzP1tPw2fM54AEjgdIAmAuCUYAg",
			"IEAMgEAgOAAMABAGAoQjjwEUAAJBAg",
			"YyAIkRlwDhQcAAAAAAAAAABEoIEO",
			"J5VTmXngoYy6CIACQKvPiAAIgIelCxc",
			"GgFIzygxOJ4x1sWHMd7zfj-pDEZMcrAYEA",
			"a4ELHg9_ByYQA5CBBMJEAQICACAO5RM",
			"ZwEIHAwAHgwGEAAMAKBAABBAfCc",
			"c393uD5gx-vPiRQeCJAFEwsYA0AC",
			"Sr6cgMMz7aADOgICIwychEEBYgSCQkCFCA",
			"RsADBgDgNBC2pDBAQChig3EBAADkpOEZTNAD",
			"OxBw3J9NpjAziQHcZRwCQwA9MgLInAAa",
			12630.941802978516
		]
	}
}
